### ISSUE TITLE
FEATURE - check container is unprivileged from inside the container (so software can configure itself accordingly)

### ISSUE THREAD

**Issue by itoffshore** [lxc/lxd#9399]
# Required information

 * Distribution: Ubuntu 
 * Distribution version: `21.10`
   * Kernel version: `5.13.0-19-generic`
   * LXC version: `4.19`
   * LXD version: `4.19`
   * Storage backend in use: `zfs`

# **FEATURE** description

I think it could be useful for software running inside an unprivileged LXD container to be able to check if it is operating in a user namespace so it can [configure itself accordingly](https://github.com/k3s-io/k3s/issues/4249).

`k3s` seems to almost work inside unprivileged LXD so this detection could also be useful for microk8s now that `KubeletInUserNamespace` is a feature gate in Kubernetes `v1.22`.

I know it is possible to check `/proc/1/environ` to see if the running system is a container, but I don't think it's currently possible inside an LXD container to check if it is privileged or unprivileged (only from outside on the host)

**Comment by stgraber** [comment]
Yeah, the definition of an unprivileged container is a container where the host root id (0) isn't mapped into the container.
